x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/IdentifierGenerator.java

```diff
@@ -55,7 +55,7 @@ public static String randomIndexPattern(Feature... features) {
             index.append("-{now/");
             index.append(ESTestCase.randomFrom("d", "M", "M-1M"));
             if (ESTestCase.randomBoolean()) {
-                index.append("{").append(ESTestCase.randomFrom("yyyy.MM", "yyyy.MM.dd")).append("}");
+                index.append("{").append(ESTestCase.randomFrom("yyyy.MM", "yyyy.MM.dd", "yyyy.MM.dd|+12:00")).append("}");
```

Comment:
The `+12:00` can be from -18 to +18 hours, I think - maybe let's have a case with a `-`, too, because that's another interesting character.